### PR TITLE
[build] Fix various spotbugs warnings 

### PR DESCRIPTION
**Pull request by nicoloboschi**
### Motivation
If you run spotbugs with JDK8 it fail with many errors.

Spotbugs with jdk11 is disabled in maven and it explains why it doesn't fail in CI
In gradle we set the reportLevel to HIGH so we haven't seen the Medium level for a while

In branch-4.14 there's still a job with a spotbugs enabled and it fails

### Changes
* Fix spotbugs warnings
   * Mostly suppressions due to https://github.com/spotbugs/spotbugs/issues/756
   * RxJava usage - call dispose of `Disposable`
   * Better synchronization on `LedgerFragmentReplicator`
* Restore spotbugs check with jdk8 and jdk11 on CI 

We'll need to cherry-pick to branch-4.14 to fix the compilation

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DirectMemoryCRC32Digest.java [hunk 1]
```diff
@@ -83,6 +83,9 @@ public void update(ByteBuf buf) {
             updateBytesMethod = CRC32.class.getDeclaredMethod("updateBytes", int.class, byte[].class, int.class,
                     int.class);
             updateBytesMethod.setAccessible(true);
+        } catch (RuntimeException e) {
```
Comment (dlg99) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DirectMemoryCRC32Digest.java:86]: let's do these changes in a separate PR for master and cherry-pick it into branch-4.14

**Review by dlg99 — COMMENTED** [review]
Let's split into two PRs (or correct me if this is not a problem in 4.14 build)

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java [hunk 2]
```diff
@@ -1238,38 +1242,42 @@ public void processResult(int rc, String s, Object ctx) {
...
-                        if (!isInteractive) {
-                            // If non interactive and force is set, then delete old data.
-                            doFormat = force;
-                        } else {
-                            // Confirm with the admin.
-                            doFormat = IOUtils
-                                    .confirmPrompt("Ledger root already exists. "
-                                            + "Are you sure to format bookkeeper metadata? "
-                                            + "This may cause data loss.");
+        return runFunctionWithMetadataBookieDriver(conf, new Function<MetadataBookieDriver, Boolean>() {
```
Comment (nicoloboschi) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java:1245]: just moved to non-lambda to add the suppression

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DirectMemoryCRC32Digest.java (hunk 1 above)
Comment (nicoloboschi) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DirectMemoryCRC32Digest.java:86]: it's okay. it should be apply cleanly on 4.14 branch. All these changes are needed to make spotbugs passes

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by eolivelli — CHANGES_REQUESTED** [review]
One test related to this change failed, please check:
```

Errors: 
Error:  org.apache.bookkeeper.verifier.BookkeeperVerifierTest.testBasic
Error:    Run 1: BookkeeperVerifierTest.testBasic »  Mismatched entry contents on entry 0 for l...
Error:    Run 2: BookkeeperVerifierTest.testBasic »  Mismatched entry contents on entry 0 for l...
Error:    Run 3: BookkeeperVerifierTest.testBasic »  Mismatched entry contents on entry 3 for l...
```

**Review by dlg99 — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by hangc0276** [comment]
@nicoloboschi I'm cherry-picking Prs to branch-4.14. There are too many conflicts in branch-4.14, would you please push a separate PR to branch-4.14? thanks a lot.
